packages/lodestar/src/chain/attestation.ts

```diff
@@ -0,0 +1,100 @@
+import assert from "assert";
+import {IAttestationProcessor, ChainEventEmitter} from "./interface";
+import {ILMDGHOST} from ".";
+import {Attestation, BeaconBlock, Hash, Slot} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {IBeaconDb} from "../db";
+import {getCurrentSlot, computeEpochAtSlot, getAttestingIndices} from "@chainsafe/eth2.0-state-transition";
+import {GENESIS_EPOCH} from "../constants";
+import {ILogger} from "../logger";
+import {hashTreeRoot, signingRoot} from "@chainsafe/ssz";
+
+type RootHex = string;
+
+export class AttestationProcessor implements IAttestationProcessor {
+  private readonly config: IBeaconConfig;
+  private db: IBeaconDb;
+  private logger: ILogger;
+  private chain: ChainEventEmitter;
+  private forkChoice: ILMDGHOST;
+  private pendingAttestations: Record<RootHex, Record<RootHex, Attestation>>;
```

Comment:
It would make more sense like `Record<BlockRootHex, Record<AttestationRootHex, Attestation>>`

Why not Record<BlockRootHex, Array<Attestation>>?

Reply:
You could consider Map instead of plain object (Record) for the utility methods.

Reply:
> Why not Record<BlockRootHex, Array>?

It's because I don't want to have duplicate pending attestations for a block. Later on I don't want to process an attestation twice when receiving a block.

Reply:
That makes sense